src/SqlFormatter.php

```diff
@@ -1085,12 +1085,11 @@ private function tokenize(string $string) : array
     /**
      * Format the whitespace in a SQL string to make it easier to read.
      *
-     * @param string $string    The SQL string
-     * @param bool   $highlight If true, syntax highlighting will also be performed
+     * @param string $string The SQL string
      *
      * @return string The SQL string with HTML styles and formatting wrapped in a <pre> tag
      */
-    public function format(string $string, bool $highlight = true) : string
+    public function format(string $string) : string
```

Comment:
since the highlighter was used only here, i was thinking to not pass it to the constructor, but here

Reply:
Now it also is used here:

https://github.com/doctrine/sql-formatter/blob/fc5b0a7421fd24980fae21550ab4a8315e864b8e/src/SqlFormatter.php#L1415-L1429

which makes it very useful to get the same string you entered in the case of `NullProvider` :joy: 

But seriously though, wouldn't it make the usage a bit cumbersome? If you want to highlight 3 SQL statements, you would have to provide a highlighter instance every time now.

Reply:
@goetas I disagree with this idea of passing a dependency through the both `format` and `highlight`. There is only one public method left that does not use it: `compress`. That method does not call any other method in the class, and depends only on the tokenizer. Maybe it should be extracted in a separate class? It's possible now that the tokenizer was extracted. In that case, would passing the highlighter in the constructor seem ok with you?

Reply:
@SenseException your opinion is welcome :)

Reply:
The reason for me suggesting this is that I see the `SqlFormatter.php` as a service. The service can be used to highlight a query displayed on a webpage, dumped to a file, or displayed via cli.

In all this cases, if the highlighter is passed as constructor parameter, you need to create a different instance of it. If you take as example symfony, it will not be possible to compile the container with the `SqlFormatter.php` as a service and then to use the same service in this different contexts.

Reply:
Looking better, there is also some discrepancy in the naming/behavior of the class.

if you call `format()`, it will not just format the SQL, but will also highlight it, so what is the propose for having this two methods?

so it seems that we have the following cases:

- format 
- highlight
- format and highlight

To me it seems that we can have:
```php
class SqlFormatter {
   function format(string $sql, Highlighter $highlighter = null) {
      $highlighter = $highlighter ?: new NullHighlighter();
     // ...
   }
   function highlight(string $sql, Highlighter $highlighter, bool $format = false) {
     // ...
   }
}
```

What do you think... (im not really happy with the proposed interface my self... but no better ideas so far)

Reply:
Using a boolean in a method signature is a known code smell, see https://phpmd.org/rules/cleancode.html , but good point about the double responsibility of format(). How about this interface?

Also, what would you return when calling `highlight` with `$format` set to `true`? The exact same as `format` with a highlighter?

> If you take as example symfony, it will not be possible to compile the container with the SqlFormatter.php as a service and then to use the same service in this different contexts.

But you can compile the container with 2 or 3 services that are all instances of `SqlFormatter`, and you could use different services in different contexts. If we add an interface, we could alias the interface to the most sensible service, which would IMO be `new SqlFormatter()` (auto detection base on `PHP_SAPI`).

My worry with your proposal is that the interface of `SqlFormatter` becomes needlessly complicated, especially for the main use case where people want both highlighting and formatting.